runtime/proto/corfu_store_metadata.proto

```diff
@@ -0,0 +1,38 @@
+syntax = "proto3";
+
+package org.corfudb.runtime;
+option java_package = "org.corfudb.runtime";
+
+import "google/protobuf/descriptor.proto";
+import "google/protobuf/any.proto";
+
+// Logical Timestamp to perform queries and transactions at a particular snapshot.
+message Timestamp {
+    int64 epoch = 1;
+    int64 sequence = 2;
+}
+
+// Table name.
+message TableName {
+    string namespace = 1;
+    string table_name = 2;
+}
+
+// Descriptor to provide self describing schemas for the records.
+message TableDescriptors {
+    google.protobuf.DescriptorProto keyDescriptor = 1;
+    google.protobuf.DescriptorProto valueDescriptor = 2;
+}
+
+// Metadata.
+message TableMetadata {
+    TableName table_name = 1;
+    bool disk_based = 2;
```

Comment:
looks like disk_based and cache should be an enum
```
enum Persistent {
  disk_based, cache
}
```

Reply:
These are 2 different options.
Disk based is for backing the Corfu tables by a disk.
Cache is for non-cacheable streams.